carma-messenger-core/cpp_message/src/EmergencyVehicleResponse_Message.cpp

```diff
@@ -0,0 +1,325 @@
+            if (str_len >= Header_constant.STATIC_ID_MIN_LENGTH && str_len <= Header_constant.STATIC_ID_MAX_LENGTH)
+            {
+                for (size_t i = 0; i < str_len; i++)
+                {
+                    sender_id += message->value.choice.TestMessage06.header.hostStaticId.buf[i];
+                }
+            }
+            else
+                sender_id = Header_constant.STRING_DEFAULT;
+
+            header.sender_id = sender_id;
+
+            // get recepient id
+            str_len = message->value.choice.TestMessage06.header.targetStaticId.size;
+            if (str_len >= Header_constant.STATIC_ID_MIN_LENGTH && str_len <= Header_constant.STATIC_ID_MAX_LENGTH)
+            {
+                for (size_t i = 0; i < str_len; i++)
+                {
+                    recipient_id += message->value.choice.TestMessage06.header.targetStaticId.buf[i];
+                }
+            }
+            else
+                recipient_id = Header_constant.STRING_DEFAULT;
+
+            header.recipient_id = recipient_id;
+
+            // get bsm id
+            // sender_bsm_id is meant to represent the vehicle BSM id in hex string (Ex: FFFFFFFF)
+            str_len = message->value.choice.TestMessage06.header.hostBSMId.size;
+            for (size_t i = 0; i < str_len; i++)
+            {
+                sender_bsm_id += message->value.choice.TestMessage06.header.hostBSMId.buf[i];
+            }
+            if (str_len < Header_constant.BSM_ID_LENGTH)
+            {
+                sender_bsm_id = std::string((Header_constant.BSM_ID_LENGTH - str_len), '0').append(sender_bsm_id);
+            }
+            else if (str_len > Header_constant.BSM_ID_LENGTH)
+            {
+                RCLCPP_WARN(node_logging_->get_logger(), "BSM ID -size greater than limit, changing to default");
+                sender_bsm_id = Header_constant.BSM_ID_DEFAULT;
+            }
+
+            header.sender_bsm_id = sender_bsm_id;
+
+            // get plan id
+            str_len = message->value.choice.TestMessage06.header.planId.size;
+            if (str_len == Header_constant.GUID_LENGTH)
+            {
+                for (size_t i = 0; i < str_len; i++)
+                {
+                    plan_id += message->value.choice.TestMessage06.header.planId.buf[i];
+                }
+            }
+            else
+                plan_id = Header_constant.GUID_DEFAULT;
+
+            header.plan_id = plan_id;
+
+            // recover uint64_t timestamp from string
+            str_len = message->value.choice.TestMessage06.header.timestamp.size;
+            timestamp = 0;
+            char timestamp_ch[str_len + 1];
+            for (size_t i = 0; i < str_len; i++)
+            {
+                timestamp_ch[i] = message->value.choice.TestMessage06.header.timestamp.buf[i];
+            }
+            timestamp_ch[str_len] = 0; // String needs a null terminator
```

Comment:
Should this be '\0'?

Reply:
Both are valid, I'll change it to '\0' to be more clear